File: server/app/schemas/subscription.py
from pydantic import BaseModel
from datetime import datetime


class CreatePlanSchema(BaseModel):
    plan_name: str
    amount: int
    validity_days: int


class PlanOut(BaseModel):
    id: str
    plan_name: str
    amount: int
    validity_days: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CreateOrderSchema(BaseModel):
    plan_id: str


class VerifyPaymentSchema(BaseModel):
    plan_id: str
    razorpay_order_id: str
    razorpay_payment_id: str
    razorpay_signature: str


class StudentSubscriptionOut(BaseModel):
    id: str
    user_id: str
    group_id: str
    plan_id: str
    amount: int
    start_date: datetime
    valid_till: datetime
    is_active: bool
    razorpay_order_id: str | None
    razorpay_payment_id: str | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: client/src/types/subscription.ts
export interface ISubscription {
  id: string;
  group_id: string;
  plan_id: string;
  amount: number;
  start_date: string;
  valid_till: string;
  is_active: boolean;
  razorpay_order_id: string;
  razorpay_payment_id: string;
  created_at: string;
  updated_at: string;
}

export interface IPlan {
  id: string;
  group_id: string;
  group_name?: string;
  plan_name: string;
  amount: number;
  validity_days: number;
  created_at: string;
}

export interface ICreatePlan {
  plan_name: string;
  amount: number;
  validity_days: number;
}

export interface IStudentGroupSubscription {
  teacher: {
    id: string;
    name: string;
    email: string;
    image_url?: string;
  };
  group: {
    id: string;
    name: string;
    description: string;
    image_url?: string;
  };
  plans: IPlan[];
  subscription: ISubscription | null;
}


export interface ITeacherSubscriptionStats {
  paid_students: number;
  total_students: number;
  total_earnings: number;
}